### PR TITLE
Implement CRON like report generation

### PR DESCRIPTION
**Pull request by gjr80**
Refer issue #17, in particular later comments.

Renamed variables/class name to remove reference to the word CRON. Located class ReportTiming (formerly class ReportCron) in reportengine.py.

### CODE REVIEW

File: bin/weewx/reportengine.py [hunk 1]
```diff
@@ -25,55 +26,92 @@
...
-    
+
     See below for examples of generators.
     """
-    
-    def __init__(self, config_dict, stn_info, gen_ts=None, first_run=True):
-        """Initializer for the report engine. 
-        
+
+    def __init__(self, config_dict, stn_info, record=None, gen_ts=None, first_run=True):
```
Comment (tkeffer) [inline bin/weewx/reportengine.py]: Would suggest adding the record parameter to the end, on the remote chance someone has called the initializer without using keywords

**Comment by tkeffer** [comment]
this looks really great! Much cleaner approach than where we started.

**Comment by tkeffer** [comment]
This will also set us up nicely for reports that need direct access to the last archive record, without going through the database.

**Comment by matthewwall** [comment]
very nicely done, gary.  i took the liberty of making a few editorial fixes to the docs - at commit ce949f2.

**Pull request merged**